htdocs/product/class/api_products.class.php

```diff
@@ -1574,6 +1575,11 @@ public function getVariants($id)
 			$prodc2vp = new ProductCombination2ValuePair($this->db);
 			$combinations[$key]->attributes = $prodc2vp->fetchByFkCombination((int) $combination->id);
 			$combinations[$key] = $this->_cleanObjectDatas($combinations[$key]);
+
+			$productModel = new Product($this->db);
+			$productModel->fetch((int) $combination->fk_product_child);
+			$productModel->load_stock();
```

Comment:
Because we want API to be a low level and this method may generate a high level of subrequest, it is better to add an option to do this.
For example, other api have a parameter call $includestockdata set to 0 by default. And load of stock is done only if this param is set to 1. Can you add this ?

Reply:
Yes, of course. I will make that change.